js/store.js

```diff
@@ -29,31 +42,32 @@ const store = {
     return null;
   },
 
-  setFirstParty(param) {
-    /**
-     * @param object with hostname as key
-     * @todo validate the param.-
-     * @todo code to be updated in next PR
-     */
-    return this.set(param);
+  async setFirstParty(domain, data) {
+    const websites = await this._read();
+    websites[domain] = data;
+
+    this._write(websites);
   },
 
-  async set(websites) {
-    /**
-      * @todo code to be updated in next PR
-    */
-    return await browser.storage.local.set({ websites });
+  async setThirdParty(parent, domain, data) {
+    const websites = await this._read();
+    const firstParty = websites[parent];
+    firstParty['thirdPartyRequests'][domain] = data;
+
+    this.setFirstParty(parent, firstParty);
   },
 
   async remove() {
     return await browser.storage.local.remove('websites');
   },
 
   addListeners() {
-    /*
-    * @todo update the code
-    */
+    browser.storage.onChanged.addListener((changes, area) => {
```

Comment:
Unless we are expecting the storage to be written from elsewhere does this actually matter?

When write calls, this will update and might actually cause a place where several consecutive writes actually overwrite the previous data.